airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -460,6 +460,45 @@ class ContactListStream(Stream):
     limit_field = "count"
 
 
+class DealStageHistoryStream(Stream):
+    """Deal stage history, API v1
+    Deal stage history is exposed by the v1 API, but not the v3 API.
+    The v1 endpoint requires the contacts scope.
+    Docs: https://legacydocs.hubspot.com/docs/methods/deals/get-all-deals
+    """
+    
+    url = "/deals/v1/deal/paged"
+    more_key = "hasMore"
+    data_field = "deals"
+    updated_at_field = "timestamp"
+    
+    def _transform(self, records: Iterable) -> Iterable:
+        for record in super()._transform(records):
+            yield {"id": record.get("dealId"), **record.get("properties", {}).get("dealstage")}
+    
+    def list(self, fields) -> Iterable:
+        params = {"propertiesWithHistory": "dealstage"}
+        yield from self.read(partial(self._api.get, url=self.url), params)
+
+
+class DealStream(CRMObjectStream):
+    """Deals, API v3"""
+    
+    def __init__(self, **kwargs):
+        super().__init__(entity="deal", **kwargs)
+        self._stage_history = DealStageHistoryStream(**kwargs)
+    
+    def list(self, fields) -> Iterable:
+        history_by_id = {}
+        for record in self._stage_history.list(fields):
+            if all(field in record for field in ("id", "dealstage")):
```

Comment:
If I am not mistaken, then this condition will never be fulfilled, because we get the `dealstage` from the `properties` and put its contents into the `record` object, so we will never have the `dealstage` key in the `record`.

Reply:
Nice catch, thank you, made a [revision](https://github.com/airbytehq/airbyte/pull/3734/files#diff-37c875e03f2e65ea1423534b0b634e18eaca797135f8d9df32360407b1d0ccb9R480)!